app/models/configured_system.rb

```diff
@@ -46,6 +46,8 @@ class ConfiguredSystem < ApplicationRecord
   scope :with_manager,                  ->(manager_id) { where(:manager_id => manager_id) }
   scope :with_configuration_profile_id, ->(profile_id) { where(:configuration_profile_id => profile_id) }
   scope :without_configuration_profile_id,          -> { where(:configuration_profile_id => nil) }
+  scope :not_ansible, -> { where.not(:type => ["ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ConfiguredSystem",
+                                               "ManageIQ::Providers::AnsibleTower::AutomationManager::ConfiguredSystem"]) }
```

Comment:
I really don't like this ☹️ ...I'll merge for now but we need a different way to do this.

cc @bdunne @agrare

Reply:
ewww

Reply:
This should really be something describing "not from automation managers", but not calling out ansible.  Or perhaps in the positive, "under configuration managers only"

Reply:
I'm tempted to actually make this a blocker...we really can't be adding references to plugins in core.  @agrare Is there a way to express this without resorting to type checks?

Reply:
Why are we excluding `AutomationManager::ConfiguredSystem`s ?

Reply:
or perhaps a way to dynamically generate the type list by asking the automation manager provider class?

Reply:
Yeah we can check if the ext_management_system is a ConfigurationManager

Reply:
@Fryguy you might know a cleaner way of checking the type but something like this would do the trick: 

`scope :configuration_manager, -> { where(:manager => ManageIQ::Providers::ConfigurationManager.all) }`

```
>> ConfiguredSystem.configuration_manager
  ConfiguredSystem Load (0.8ms)  SELECT  "configured_systems".* FROM "configured_systems" WHERE "configured_systems"."manager_id" IN (SELECT "ext_management_systems"."id" FROM "ext_management_systems" WHERE "ext_management_systems"."type" IN ('ManageIQ::Providers::ConfigurationManager', 'ManageIQ::Providers::Foreman::ConfigurationManager')) LIMIT $1  [["LIMIT", 11]]
  ConfiguredSystem Inst Including Associations (0.1ms - 1rows)
=> #<ActiveRecord::Relation [#<ConfiguredSystem id: 1, type: nil, hostname: nil, direct_operating_system_flavor_id: nil, configuration_profile_id: nil, manager_id: 7, manager_ref: nil, created_at: "2020-04-23 16:29:11", updated_at: "2020-04-23 16:29:11", last_checkin: nil, build_state: nil, configuration_location_id: nil, configuration_organization_id: nil, ipaddress: nil, mac_address: nil, ipmi_present: nil, puppet_status: nil, direct_customization_script_ptable_id: nil, direct_customization_script_medium_id: nil, operating_system_flavor_id: nil, customization_script_medium_id: nil, customization_script_ptable_id: nil, inventory_root_group_id: nil, virtual_instance_ref: nil, counterpart_type: nil, counterpart_id: nil>]>
```

Reply:
> Why are we excluding AutomationManager::ConfiguredSystems ?

Technically the configured_systems records include records from both ConfigurationManagers as well as AutomationManagers.  When viewing the list form the Configuration -> Configured Systems we need to exclude them.

Now that I've written that out, that is EXACTLY what we do with Cloud instances/teplates vs Infra vms/templates, so whatever we do there is exactly that we should do here.  @h-kataria How do we do that differentiation for Cloud/Infra vms?

Reply:
> `scope :configuration_manager, -> { where(:manager => ManageIQ::Providers::ConfigurationManager.all) }`

Minot, but I think I'd changed that slightly to 

`scope :under_configuration_managers, -> { where(:manager => ManageIQ::Providers::ConfigurationManager.all) }`

Otherwise, `:configuration_manager` appears like a has_one relationship and would return a configuration_manager, but it doesn't

Reply:
>  When viewing the list from the Configuration -> Configured Systems we need to exclude them.

That makes much more sense, we want to group them not exclude some

> Minor, but I think I'd changed that slightly to

Oh yeah wasn't suggesting a name haha